docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
+
+### The overall Sequence Diagram:
+![Sequence Diagram](assets/images/sequenceDiagram.png)    
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110598041/4d920459-5c65-407a-92e9-23c05c055f69)
does not return function control back to the caller, same goes for a few other function calls in the same sequence diagram